### DISCUSSION TITLE
Sending video from server to server

### DISCUSSION THREAD

**Discussion by Tormund1985** [bluenviron/mediamtx#3101] [Questions]
### Question

I am going to present you with a problem and see if you can help me make the best decision.

I'm developing a multi-camera motion detection and recording station, including smartphones, raspberry, microcontrollers...

There are computers that I have them under a given domain (no-ip, duckdns, ngrok...)  or even through a static ip, but in others it is not possible to install a domain. 

Is there any option so that from a raspberry with mediamtx I can send the stream rtsp/rtmp or other to the server and not capture it from the server (requiring to know the url of the stream through ip static or domain...). 

Currently I have a Shinobi server to which I send from smartphones the images from mobiles and capture other devices with domains and static ip, but from the raspberry and microcontrollers it is not possible.

Thanks in advance.

**Comment by aler9** [comment]
Hello, if you want to forward a stream from a server without a fixed IP or domain, to another server, which has a fixed IP or domain,  instructions are in the "forward streams" section of the README:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#forward-streams-to-other-servers

Otherwise if you want to forward a stream from a server which has a fixed IP or domain, to a server without a fixed IP or domain, instructions are in "RTSP cameras and servers" section of the README:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#rtsp-cameras-and-servers

**Comment by Tormund1985** [comment]
> Hello, if you want to forward a stream from a server without a fixed IP or domain, to another server, which has a fixed IP or domain, instructions are in the "forward streams" section of the README:
> 
> https://github.com/bluenviron/mediamtx?tab=readme-ov-file#forward-streams-to-other-servers
> 
> Otherwise if you want to forward a stream from a server which has a fixed IP or domain, to a server without a fixed IP or domain, instructions are in "RTSP cameras and servers" section of the README:
> 
> https://github.com/bluenviron/mediamtx?tab=readme-ov-file#rtsp-cameras-and-servers

Thank you very much for your response in advance.

Let's see if I understand, I on my raspberry could set the following configuration to send the video stream to a server hosted at rtsp://myserver:8554?

pathDefaults:
  runOnReady: >
    ffmpeg -i rtsp://localhost:554/mycsicamera
    -c copy
    -f rtsp rtsp://myserver:8554/raspberrycam
  runOnReadyRestart: yes

And on my server (myserver) I would receive it with the following configuration:

pathDefaults:
  runOnReady: >
    ffmpeg -i rtsp://user:pass@localhost:8554/raspberrycam
    -c copy
    -f rtsp rtsp://localhost:9554/raspberrycam01
  runOnReadyRestart: yes

It would be as simple as this?

**Reply by aler9** [reply]
no, it's much simpler. You don't need any configuration on myserver.

[a comment by an automated account is not shown]
